tornado_botocore/base.py

```diff
@@ -84,26 +83,50 @@ def _send_request(self, request_dict, operation_model, callback=None):
 
         req_body = getattr(request.body, 'buf', request.body)
 
-        request = HTTPRequest(
-            url=request.url,
-            headers=request.headers,
-            method=request.method,
-            body=req_body,
-            validate_cert=False,
-            proxy_host=self.proxy_host,
-            proxy_port=self.proxy_port,
-            connect_timeout=self.connect_timeout,
-            request_timeout=self.request_timeout
-        )
+        kwargs = {
+            "url":request.url,
+            "headers":request.headers,
+            "method":request.method,
+            "validate_cert":False,
+            "proxy_host":self.proxy_host,
+            "proxy_port":self.proxy_port,
+            "connect_timeout":self.connect_timeout,
+            "request_timeout":self.request_timeout
+        }
 
         if callback is None:
             # sync
+            if hasattr(req_body, 'read'):
+                kwargs["body"]=req_body.read()
+            else:
+                kwargs["body"]=req_body
+
+            request = HTTPRequest(**kwargs)
+
             return self._process_response(
                 HTTPClient().fetch(request),
                 operation_model=operation_model
             )
 
         # async
+
+        if hasattr(req_body, 'read'):
+            @gen.coroutine
+            def producer(write):
+                while True:
+                    chunk = req_body.read(4 * 1024)
+                    if not chunk:
+                        # Complete.
+                        break
+                    yield write(chunk)
+            req_body.seek(0)
+            kwargs["body_producer"]=producer
+            self._fetch_coroutine(HTTPRequest(**kwargs), operation_model, callback)
+            return
```

Comment:
Note we could just .read the entire stream here instead of using a body_producer... But presumably that read would be blocking since we're passing the image as a stream directly from this other async request here:

https://github.com/2hat/pottymouth/pull/1881/files#diff-da7d53f75a9f7a5534aa4597832df43b523a8728b6fb21b1ac1653a42af84ae1R144-R145